base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ActionsViewTable.java

```diff
@@ -223,84 +225,82 @@ public void addEditingSupport(StringEditingSupport<ScriptGeneratorAction> editin
+	 * @param columns The columns of the table to compare the action to the current display against.
 	 */
-	@Override
-	public void setRows(Collection<ScriptGeneratorAction> rows) {
-		if (!viewer.getTable().isDisposed()) {
-			int focusRow = getSelectionIndex();
-			ScriptGeneratorAction previousSelection = firstSelectedRow();
-			int focusColumn = 0;
-			
-			if (shiftCellFocusToNewlyAddedRow) {
-				focusRow = viewer.getTable().getSelectionIndex() + 1;
-			} else if (focusRow != -1) {
-				// When no focus is selected getFocusCell returns null
-				var focusCell = Optional.ofNullable(viewer.getColumnViewerEditor().getFocusCell());
-				if (focusCell.isPresent()) {
-					focusColumn = focusCell.get().getColumnIndex();
-				}
-			}
-			
-			viewer.setInput(new WritableList<ScriptGeneratorAction>(rows, null));
-			
-			if (selectedRows().size() == 1) {
-				// If the action on the specified row has changed then don't return focus to it
-				if (previousSelection.equals((ScriptGeneratorAction) viewer.getElementAt(focusRow)) || shiftCellFocusToNewlyAddedRow) {
-					setCellFocus(focusRow, focusColumn);
-					if (!shiftCellFocusToNewlyAddedRow) {
-					    // Fixes issue see in https://github.com/ISISComputingGroup/IBEX/issues/5708 (hopefully temporary)
-					    Optional<Integer> editingSupportFocusColumn = getEditingSupportFocusColumnFromViewerTableFocusColumn(focusColumn);
-						editingSupportFocusColumn.ifPresent(
-							offsetFocusColumn -> editingSupports.get(offsetFocusColumn).resetSelectionAfterFocus()
-						);
-					}
-					shiftCellFocusToNewlyAddedRow = false;
-				}
+	private void updateAction(ScriptGeneratorAction action, int index, TableColumn[] columns) {
+		try {
+			var item = viewer.getTable().getItem(index);
+			if (valuesDiffer(item, columns, action)) {
+				viewer.replace(action, index);
 			}
+		} catch (IllegalArgumentException e) {
+			// Thrown if we cannot get them item from the table as it doesn't exist so we need to add it
+			viewer.add(action);
 		}
-	} 
+	}
+	
 	/**
-	 * Sets Rows without touching focus, called when updating globals so table is updated but focus
-	 * not changed.
+	 * Update the actions being displayed by this table.
+	 * 
+	 * @param newActions The actions to display.
 	 */
-	public void setRowsNoFocus(Collection<ScriptGeneratorAction> rows) {
-		viewer.setInput(new WritableList<ScriptGeneratorAction>(rows, null));
+	public void updateActions(List<ScriptGeneratorAction> newActions) {
+		removeDeletedRows(newActions);
+		var columns = viewer.getTable().getColumns();
+		for (int i = 0; i < newActions.size(); i++) {
+			updateAction(newActions.get(i), i, columns);
+		}
+		this.actions = newActions;
 	}
+	
 	/**
 	 * Sets focus of cell.
 	 * @param row row number of table
 	 * @param column column number of table
 	 */
 	public void setCellFocus(int row, int column) {
-		if (row >= 0) {
-			viewer.editElement(viewer.getElementAt(row), column);
+		var element = viewer.getElementAt(row);
```

Comment:
Fixes NullPointerException where the element is null so editElement cannot handle it.